modules/store/src/action_creator.ts

```diff
@@ -133,7 +134,10 @@ export function union<
   return undefined!;
 }
 
-function defineType(type: string, creator: Creator): Creator {
+function defineType<T extends string>(
+  type: T,
+  creator: TypedCreator<T>
```

Comment:
Again, as explained later, `ActionCreator<T>` is also not fitting as it is defined since the inner `Creator` does not match type. If you look at how we call `defineType`, you'll see that e.g. function `() => ({ type })` is explicitly in contradiction with `Creator` type as return type contains `type` property.